assets/src/editor/events/ticket-assignments-manager/hooks/use-counts-manager.js

```diff
@@ -0,0 +1,118 @@
+/**
+ * External imports
+ */
+import { useRef, useMemo, useCallback } from '@wordpress/element';
+import { reduce } from 'lodash';
+import { __ } from '@eventespresso/i18n';
+
+// @todo, when the wp.data.useSelect hook is available, we can utilize that in
+// this custom hook and then we'll be able to remove the withSelect hoc usage
+// in the composed `TicketAssignmentsManagerModal` component.
```

Comment:
It would greatly simplify things if we were able to say that the new ticket/date editor will _require_ WordPress 5.3 (which assuming everything goes to plan, will be the version containing the new `useDispatch` and `useSelect` hooks).  So I'm wondering if we should bite the bullet and start developing the editor with the Gutenberg plugin active so we can use those hooks and simplify the code now.

Pros:

- Avoid the current HOC hell we have now (only using them in very necessary places)
- Simplifies the logic flow for a lot of the big important things such as TAM.

Cons:

- we create a hard dependency on a future api that is stable in GB but hasn't landed in WP core yet.  So there's uncertainty there.
- We _will_ need to bump minimum WP requirements to a more recent version (likely WP 5.3+) when this lands.
- We need to remember to develop with the GB plugin active while working on this.
- Some more mental decision angst around whether we build something in a HOC or using new hooks for things we write.

I just bring this up, not because I think the decision needs to be made in this  pull, but as something to keep in mind and think on.  I haven't convinced myself one way or the other yet so we can sit on it.  But I'm sure this will come up again as I continue to go through and improve the logic of the new editor.

Reply:
I'm almost always for going with the more future forward way of doing things and here's some of the reasons that come to mind for me:

- there's already a hard dependency on a fairly recent version of WP core anyways, so I don't think bumping that up to 5.3 is that big of a deal.
- yes there's a risk that using the new `useDispatch` and `useSelect` hooks may force us to refactor things **IF** those APIs change prior to release, but not doing so and using the existing API means more convoluted code that we are likely going to want to refactor one day anyways. So _potential_ refactors in exchange for easier to deal with code versus guaranteed technical debt and more obtuse code is an easy choice for me.
- activating a plugin for dev purposes may actually prove to be too hard... LOLZ j/k
- more "mental decision angst around whether we build something in a HOC or using new hooks" is nothing compared to the mental angst involved with what to name things, so this isn't a big deal for me either. I think I will likely favour hooks and functional composition over HOCs even though I haven't actually started using them yet.